processor/memorylimiterprocessor/memorylimiter_test.go

```diff
@@ -440,3 +109,221 @@ func TestNoDataLoss(t *testing.T) {
 	err = processor.Shutdown(context.Background())
 	require.NoError(t, err)
 }
+
+// TestMetricsMemoryPressureResponse manipulates results from querying memory and
+// check expected side effects.
+func TestMetricsMemoryPressureResponse(t *testing.T) {
+	md := pmetric.NewMetrics()
+	ctx := context.Background()
+
+	tests := []struct {
+		name     string
+		mlCfg    *Config
+		getMemFn func() (uint64, error)
+		hasError bool
+	}{
+		{
+			name: "fixed_limit",
+			mlCfg: &Config{
+				CheckInterval:       1 * time.Millisecond,
+				MemoryLimitMiB:      2,
```

Comment:
Why doesn't it fail with such low limits?

Reply:
The test starts over the soft limit. But the checkMemLimits() also does a GCing, which free up enough space to not trigger the mem limit.

Reply:
previous scenarios are replaced so no longer is relevant